ui/src/components/SecretForm/index.jsx

```diff
@@ -132,6 +133,7 @@ export default class SecretForm extends Component {
   render() {
     const { classes, isNewSecret, loading } = this.props;
     const { secretName, editorValue, expires, showSecret } = this.state;
+    const isSecretDirty = !equals(secret, this.props.Secret);
```

Comment:
This won't work because the secret values are actually in `this.state.secretName`, `this.state.expires` and `this.state.editorValue`. Also, `this.props.Secret` should be `this.props.secret` (lower case "s").

To fix this, you would need to do something like:

```jsx
const { secret, classes, isNewSecret, loading } = this.props;
const { secretName, editorValue, expires, showSecret } = this.state;
const isSecretDirty =
  secretName !== secret.name ||
  editorValue !== safeDump(secret.secret) ||
  expires !== secret.expires;
```